Add docs for opensafely launch (#1717)
opensafely/documentation: docs/opensafely-cli.md

This can be useful if you want to explore the image manually.

### `launch` - Run web based interactive tools

> Many users may prefer to use some of the interactive development environments that are commonly used for different languages, like Rstudio or Jupyter Lab.

Reviewer: ```suggestion
Many users may prefer to use some of the interactive development environments that are commonly used for different languages, like RStudio or JupyterLab.
```

> Using `opensafely launch`, you can run these tools from the OpenSAFELY images, accessing them via your browser. This means that code you write and run in these tools will be running the OpenSAFELY provided environment, with the right dependencies and versions, rather than what ever you have installed on your machine.

Reviewer: ```suggestion
Using `opensafely launch`, you can run these tools from the OpenSAFELY images, accessing them via your browser. This means that code you write and run in these tools will be running the OpenSAFELY-provided environment, with the right dependencies and versions, rather than whatever you have installed on your own machine.
```

> The supported tools of Rstudio and Jupyter Lab are documented below, but they share a common set of options, which you can see with

Reviewer: ```suggestion
The supported tools of RStudio and JupyterLab are documented below, but they share a common set of options, which you can see with
```

```
opensafely launch --help
```

#### `opensafely launch rstudio` - Running RStudio Server

> This sub-command provides the RStudio interface to the r image. This lets you run your code in RStudio using the same versions of R and all the packages that are the OpenSAFELY secure backend.

Reviewer: ```suggestion
This sub-command provides the RStudio interface to the R image. This lets you run your code in RStudio using the same versions of R and all the packages that are available in the OpenSAFELY secure backend.
```

> Note that if you are using a Github [Codespaces](/getting-started/how-to/use-github-codespaces-in-your-project/#how-to-access-rstudio), then this is run automatically for you.

Reviewer: ```suggestion
Note that if you are using a Github [Codespace](/getting-started/how-to/use-github-codespaces-in-your-project/#how-to-access-rstudio), then this is run automatically for you.
```

To launch an RStudio Server session please navigate to your research repository and run

```bash
opensafely launch rstudio
```

This will launch an RStudio Server session, and open it in a browser window. A message in your terminal will tell you which port it is running on, in case you need to know.

By default it will use the most recent version of the OpenSAFELY R image, but you can specify an older version with:

```bash
opensafely launch rstudio:v1
```

In your RStudio Server session you can work on your files as usual, including saving your edits.

If you have an `.Rproj` file at the top level of your repository, the command will detect this and will open the RStudio Server session in RStudio's project mode. In this mode the Git pane is available, so you can make commits, or you can make commits locally as you usually do.

To end your RStudio Server session press ++ctrl+c++ in the Terminal window in which you ran the `opensafely rstudio` command.

> Note that the `opensafely` CLI is not vendored within the RStudio image. Hence to use CLI commands such as `opensafely run`, please do so from a local Terminal session and not from the Terminal pane in your RStudio Server session. Whereas, the `opensafely` CLI is vendored within the Codespaces version of RStudio Server, and so you can use the RStudio Terminal in this case.

Reviewer: ```suggestion
Note that the `opensafely` CLI is not available within the RStudio image. Hence to use CLI commands such as `opensafely run`, please do so from a local Terminal session and not from the Terminal pane in your RStudio Server session. If you are using a Codespace, the `opensafely` CLI **is** availablle within the RStudio Server session, and so you can use the RStudio Terminal in this case.
```

`http://localhost`… — and use that URL in a web
browser to access JupyterLab.

> By default, it will run Jupyter Lab in the most recent version of the python image, but you can to run an older version with:

Reviewer: ```suggestion
By default, it will run JupyterLab in the most recent version of the python image, but you can to run an older version with:
```

```bash

> opensafely launch juypter:v1

Reviewer: ```suggestion
opensafely launch jupyter:v1
```
